src/utils/etheroll-contract.js

```diff
@@ -15,65 +15,73 @@ const etherscanUrls = {
   [Networks.ropsten]: 'https://ropsten.etherscan.io',
 };
 
-function getProfit(betSize, chances) {
-  let profit = 0;
+
+function getPayout(betSize, winningChances) {
+  return 100 / winningChances * betSize;
+}
+
+function cutHouseEdge(payout) {
   const houseEdge = 1 / 100.0;
-  const chancesLoss = 100.0 - chances;
-  if (chances !== 0 && chancesLoss !== 0) {
-    let payout = ((chancesLoss / chances) * betSize) + betSize;
-    payout *= (1 - houseEdge);
-    profit = payout - betSize;
+  return payout * (1 - houseEdge);
+}
+
+function getProfit(betSize, winningChances) {
+  if (winningChances === 0) {
+    return 0;
   }
-  return profit;
+  const rawPayout = getPayout(betSize, winningChances);
+  const netPayout = cutHouseEdge(rawPayout);
+
+  return Math.max(netPayout - betSize, 0);
+}
+
+
+// Merges bet logs (LogBet) with bet results logs (LogResult).
+function mergeLogs(logBetEvents, logResultEvents) {
+  const findLogResultEventBylogBetEvent = logBetEvent => (
+    logResultEvents.find(logResultEvent => (
+      logResultEvent.args.BetID === logBetEvent.args.BetID
+    ))
+  );
+
+  return logBetEvents.map(logBetEvent => ({
+    logBetEvent,
+    logResultEvent: findLogResultEventBylogBetEvent(logBetEvent),
+  }));
```

Comment:
Don't we increase the time complexity here?
Before we had two operations called sequentially `reduce()` and `map()` which AFAIK are `O(n)` (well `f(n)`):

1. logResultEvents.reduce() 
2. logBetEvents.map()

Within the map we were accessing a key via `logResultEvent: logResultEventsDict[logBetEvent.args.BetID]` which is `O(1)`.
So in total `mergeLogs()` was `2*O(n)` to me.

But in the new implementation we do a `findLogResultEventBylogBetEvent(logBetEvent)` a `O(n)` within the `logBetEvents.map()` which is `O(n)`.
Hence it feels like the `mergeLogs()` would be a `O(n²)` correct?
Probably/hopefully not a big deal, but worth noting